ui/src/importer/importerTaskDetails/ImportedDocuments.js

```diff
@@ -1,194 +1,69 @@
-import _get from 'lodash/get';
 import _cloneDeep from 'lodash/cloneDeep';
-import { CancelImportTask } from './cancelImportTask';
-import { RenderStatistics } from './ImportStats';
-import { ImportedTable } from './ImportedTable';
-import { ImportedSearch } from './ImportedSearch';
-import { CdsBackOfficeRoutes } from '../../overridden/routes/BackofficeUrls';
-import { Link } from 'react-router-dom';
-import { importerApi } from '../../api/importer';
-import { invenioConfig } from '@inveniosoftware/react-invenio-app-ils';
+import { RenderStatistics } from './ImporterTaskDetailsComponents/ImportStats';
+import { ImportedTable } from './ImporterTaskDetailsComponents/ImportedTable';
+import { ImporterReportHeader } from './ImporterTaskDetailsComponents/ImporterReportHeader';
+import { NotFound } from '@inveniosoftware/react-invenio-app-ils';
+import { ImporterReportStatusLabel } from './ImporterTaskDetailsComponents/ImporterReportStatusLabel';
+import { ImporterReportMode } from './ImporterTaskDetailsComponents/ImporterReportMode';
 
 export class ImportedDocuments extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      importCompleted: false,
-      data: null,
-      isLoading: false,
       selectedResult: 'records',
       searchText: '',
       activePage: 1,
-      statistics: {
-        records: {
-          text: 'Records',
-          value: '',
-          filterFunction: record => record,
-        },
-        records_created: {
-          text: 'Records created',
-          value: 0,
-          filterFunction: record => record.action === 'create',
-        },
-        records_deleted: {
-          text: 'Records deleted',
-          value: 0,
-          filterFunction: record => record.action === 'delete',
-        },
-        records_updated: {
-          text: 'Records updated',
-          value: 0,
-          filterFunction: record => record.action === 'update',
-        },
-        records_with_errors: {
-          text: 'Records with errors',
-          value: 0,
-          filterFunction: record => _isNull(record.action),
-        },
-        records_with_item: {
-          text: 'Records with eItem',
-          value: 0,
-          filterFunction: record => !_isNull(record.eitem),
-        },
-        records_with_serials: {
-          text: 'Records with Serials',
-          value: 0,
-          filterFunction: record => !_isEmpty(record.series),
-        },
-      },
-      filteredRecords: null,
     };
-    this.mode = '';
-    this.status = '';
-    this.lastIndex = 0;
-  }
 
-  componentDidMount() {
-    const { taskId } = this.props;
-    this.intervalId = setInterval(
-      () => this.checkForData(taskId),
-      invenioConfig.IMPORTER.fetchTaskStatusIntervalSecs
-    );
-    this.checkForData(taskId);
+    this.lastIndex = 0;
+    this.statistics = {
+      records: {
+        text: 'Records',
+        value: '',
+        filterFunction: record => record,
```

Comment:
You are skipping using the filterFunction when you have all the records (which is right), so this line should be deleted ( I can do it in next PR so we merge ASAP).

Reply:
Nevermind, I just saw you are using it to calculate the filteredRecords. You only are not using it in the statistics.